### ISSUE TITLE
More consistent and descriptive file and folder names for backups

### ISSUE THREAD

**Issue by redactedscribe** [MikeMaximus/gbm#272]
Although I understand that this feature was added in v1.3.1 to prevent accidental overwrites, I've had the global _Create a sub-folder for each game_ option enabled for some versions now so that I could manually browse the backups (useful if browsing the backups in cloud storage). With Game IDs for folder names, it impossible to make sense of things outside of GBM without opening up the archives and checking the contents.

Preferably, folders would contain the game configuration's name with its Game ID appended, thereby making any duplicate game backups always unique files / folders, whilst still being human readable. As seen in #271, it seems others would like this too.

Example naming schemes:

```
Folders (<name> <game_id>):

Cyberpunk 2077 [f8cb08d5-39d5-4b6d-92f1-9029ed49278b]/
Cyberpunk 2077 - f8cb08d5-39d5-4b6d-92f1-9029ed49278b/


Files (<name> <backup_timestamp> <game_id>):

Cyberpunk 2077 7-14-2022-00-00-00 [f8cb08d5-39d5-4b6d-92f1-9029ed49278b].7z
Cyberpunk 2077 7-14-2022-00-00-00 - f8cb08d5-39d5-4b6d-92f1-9029ed49278b.7z
```

This kind of naming scheme would mean the backup folder wouldn't end up being a mix of readable names and Game IDs (UUIDs) when _Use Game ID for folder and file names_ isn't enabled. The option could be removed altogether in fact.

It should be easy enough to determine the Game ID from a file / folder with a bit of regex if needed, or even just by reading the last 36 characters if the ID is appended plainly. Typically the date would be appended last to chronologically sort files, but since a Game ID shouldn't ever change, the ordering would still be determined by the date if present.

The reason for suggesting the Game ID to be last is that it's the least relevant bit of info when manually browsing the backup folder, and so a game's name and backup date are more likely to be visible without needing to increase window size / column width to see both.

Also, **is there a good reason why a backup that isn't set up to do multiple backups is named without a date in its name?** I think that's quite useful information which is sometimes lost on file transfers / downloads.

And, **why not use the same naming scheme regardless of the situation?** An archive for example can either be named as the game name, or the game name and a backup timestamp, or just its UUID. Something universal, informative, and readable would be preferable in my opinion.

Thanks.

GBM 1.3.1
Windows 11

**Comment by MikeMaximus** [comment]
One consistent naming scheme is a good idea and will simply a lot of things.

I don't particularly like the UUID names either, but I didn't want to dig too much into code that re-test everything last release.

**Comment by Darthagnon** [comment]
Wait, did v1.3.1 start adding the UUID to game files/folders 😥? (I'm still on v1.3.0)

**Comment by MikeMaximus** [comment]
1.3.1 will use the UUID for files and folder names when a configuration is using a duplicate name.  Prior 1.3.1, configurations with duplicate names would just overwrite each other.

**Comment by MikeMaximus** [comment]
I've made the following changes in v1.3.2 for this issue and #271.  

- The format "Cyberpunk 2077 [f8cb08d5-39d5-4b6d-92f1-9029ed49278b]" will automatically be used for folder and file names of configurations using duplicate names.
- GBM is now more strict with file name lengths to minimize the chances of hitting the default Windows path cap.
- The "Use Game ID for folder and file names" option has been removed from Settings.

I've decided against going all in on a single naming scheme at this time.
